### ISSUE TITLE
pg_query_stats normalization

### ISSUE THREAD

**Issue by Dulanic** [ankane/pghero#381]
Hello,

Have you considered normalizing pg_query_stats? I would like to keep query stats longer, but the tables starts getting massive. The same queries are recorded thousands if not dozens of thousands of times. Some of these queries are long and take up significant space over time.

**Comment by ankane** [comment]
Hey @Dulanic, thanks for the suggestion. I think it'll add more complexity, but will consider it for PgHero 3.0 (#318). Someone could also put together a script that does this before cleaning up previous stats.

**Comment by Dulanic** [comment]
Thanks @ankane. I did this testing, and it reduced my query table by over 80% and that doesn't consider the hash column that I forgot about until now. That shouldnt be as large of a impact. This was my test code, forgive the lazy ugly code.

```

CREATE TABLE IF NOT EXISTS public.pghero_queries
(
    query_id integer NOT NULL DEFAULT nextval('pghero_queries_query_id_seq'::regclass),
    query text COLLATE pg_catalog."default",
    CONSTRAINT pghero_queried_pkey PRIMARY KEY (query_id)
);

insert into public.pghero_queries (query)
select distinct query from public.pghero_query_stats1;

ALTER TABLE public.pghero_query_stats1
    ADD COLUMN query_id integer;
	
	
update public.pghero_query_stats1 a
set query_id =  b.query_id
from   public.pghero_queries b
 where a.query = b.query

ALTER TABLE public.pghero_query_stats1 DROP COLUMN query; 

ALTER TABLE public.pghero_query_stats1
    ADD CONSTRAINT fk_queries FOREIGN KEY (query_id)
    REFERENCES public.pghero_queries (query_id) ;

```

**Comment by ankane** [comment]
Added to #415 (as something to consider still), so closing this out.

**Comment by ankane** [comment]
Hi @Dulanic, made this change in the commit above, and it'll be included in 4.0. Thanks again for the suggestion.
